### PR TITLE
feat(core): add aliases to mitigate context issues

### PR DESCRIPTION
**Pull request by ricokahler**
### Description

This pull request enhances the `sanity` React library's build tooling by adding comprehensive comments and JSDoc annotations to improve maintainability and understanding of the code. Specifically, it elaborates on package exports, conditions, and the `getAliases` function. Additionally, it completes the associated test file to ensure the `getAliases` function operates correctly in both monorepo and normal build scenarios.

**Key changes:**
- Added detailed comments and JSDoc annotations to `aliases.ts` to explain the purpose and functionality of the code.
- Completed the test file `aliases.test.ts` with test cases for normal builds and monorepo builds.
- Mocked monorepo development aliases in the test to ensure accurate alias resolution.

**Why these changes were introduced:**
- To prevent multiple context errors due to multiple instances of the library.
- To improve code readability and maintainability.
- To ensure comprehensive testing coverage for the alias resolution functionality.

### What to review

- Review the added comments and JSDoc annotations in `aliases.ts` for accuracy and clarity.
- Review the test cases in `aliases.test.ts` to ensure they cover the expected scenarios and edge cases.
- Ensure that the monorepo aliases are correctly mocked and resolved in the tests.

### Testing

- Added test cases for both normal builds and monorepo builds in `aliases.test.ts`.
- Verified that the dynamic resolution logic in the tests mirrors the resolution logic used in `getAliases`.
- Testing manually by packing and linking this in the admin studio.

Demo:

https://github.com/user-attachments/assets/a66fc0a2-cc63-4fcb-9bcd-be06feabdf7f

### Notes for release

- Adds aliases to internal vite config used by `sanity build` and `sanity deploy` to prevent missing React context issues.

### CODE REVIEW

**Comment by rexxars** [comment]
Tests seem to be failing, will review when they pass!

File: packages/sanity/src/_internal/cli/server/__tests__/aliases.test.ts [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+    expect(browserCompatibleSanityPackageSpecifiers).toHaveLength(7)
+
+    for (const specifier of browserCompatibleSanityPackageSpecifiers) {
+      expect(currentSpecifiers).toContain(specifier)
+    }
+  })
+})
+
+describe('getAliases', () => {
```
Comment (cngonzalez) [inline packages/sanity/src/_internal/cli/server/__tests__/aliases.test.ts:48]: My personal preference is not to repeat the business logic and use shared utils in the test itself, but rather use something more aligned with a fixture (in this case, an array of `[{find: regex, replacement: path}]`) to ensure expected results and to make it a bit easier on a dev coming in trying to figure out what to expect.

With our dynamic builds and CI however, I'm not sure how feasible this is, so feel free to ignore this comment 😅

**Comment by cngonzalez** [comment]
These changes generally make sense to me and I'm sure work smoothly, but I'd feel more comfortable with someone more familiar with our Vite config having a look as well 🙇

File: packages/sanity/src/_internal/cli/server/__tests__/aliases.test.ts (hunk 1 above)
Comment (ricokahler) [reply packages/sanity/src/_internal/cli/server/__tests__/aliases.test.ts:48]: CleanShot 2024-07-18 at 19 28 05@2x
I tried to rewrite this test to be more of an integration test that actually calls vite.build but our jest environment is not having it. for now i think it's okay to merge this given i've tested it manually but i agree that this test could be improved. i've added a TODO comment for now

**Review by cngonzalez — DISMISSED** [review]

**Review by cngonzalez — APPROVED** [review]
